lib/pbench/server/api/resources/users_api.py

```diff
@@ -473,27 +491,42 @@ def delete(self, target_username):
         """
         try:
             target_user = self.get_valid_target_user(target_username)
-            if not target_user:
+        except Exception:
+            self.logger.exception(
+                "Exception occurred in DELETE request from user {!r} while validating target user {!r}",
+                self.auth.token_auth.current_user().username,
+                target_username,
+            )
+            abort(HTTPStatus.INTERNAL_SERVER_ERROR, message="INTERNAL ERROR")
+        if not target_user:
+            if self.auth.token_auth.current_user().is_admin():
                 self.logger.warning(
-                    "User {} is not authorized to delete user {}.",
+                    "User {} performed delete operation but user {} is not found.",
                     self.auth.token_auth.current_user().username,
                     target_username,
                 )
-                abort(HTTPStatus.FORBIDDEN, message="Not authorized to delete user")
-            if target_user.is_admin():
-                self.logger.warning(
-                    "Admin user is not allwed to self delete via API call. Username: {}",
-                    target_username,
+                abort(
+                    HTTPStatus.NOT_FOUND, message=f"User {target_username} not found",
                 )
-                abort(HTTPStatus.FORBIDDEN, message="Not authorized to delete user")
+            self.logger.warning(
+                "User {} is not authorized to delete user {}.",
+                self.auth.token_auth.current_user().username,
+                target_username,
+            )
+            abort(HTTPStatus.FORBIDDEN, message="Not authorized to delete user")
+        if target_user.is_admin():
+            self.logger.warning(
+                "Admin user is not allwed to self delete via API call. Username: {}",
```

Comment:
"Admin user" implies "admin" account; this is more about "User with administrative access" or we could say "User with Admin role". Also, "allowed" is missing a letter. Granted, this is an internal log message, but we won't be the only ones to read system logs and we should be a little careful both about context and security. (Some customers, for example, might not want to give Red Hat their system logs for debugging issues if the logs contain internal information like hostnames or usernames.)

Reply:
"User with ADMIN role" would certainly be clearer.  I would omit the "via API call":  that will be self-evident to the recipient of the message, and any apparent discrepancy between API policy and CLI policy is unlikely to be evident to anyone but us (and possibly the people administering the deployment).

> we won't be the only ones to read system logs and we should be a little careful both about context and security. (Some customers, for example, might not want to give Red Hat their system logs for debugging issues if the logs contain internal information like hostnames or usernames.)

This is an unsettling conundrum for me.

As a developer, I will want to know details about the context of an error or failure.  The username associated with the request and the username associated with the target of the request seem like key pieces of information.  On the server side, we will be the only ones reading the log, at least until a partner deploys their own server.  If that happens, and if they are reluctant to share their usernames (which seems...overly cautious, to me), the log is just a text file which is easily cleansed with a `sed` script...so, I would be inclined to suggest that logging the information provides much more value than it imposes in risk or cost.

Still, Dave, your point is well-taken.  (I still remember when VMS was logging failed login attempts to the system console, including the contents of the password field, which, in the days of paper logs, meant that your typos were there for anyone to see as they walked through the lab....)  But, while we shouldn't be logging security credentials, I'd like to believe that the username is an identifier and not, as such, a credential.

Reply:
> Some customers, for example, might not want to give Red Hat their system logs for debugging issues if the logs contain internal information like hostnames or usernames.

This is a concern that needs to be addressed separately and carefully, I think by merely changing the log message may not be sufficient to address this point. This warrants a separate discussion and a PR.

Reply:
Customers are often paranoid about internal identification, and I suspect this is something we'll need to deal with. It's easier to fix "from the beginning" than to retrofit later. It's also true that, especially in our initial internal deployments, we'll have a much easier time with too much information than too little; but we should be thinking about this carefully, not just throwing things out because we can.